enterprise/models.py

```diff
@@ -321,6 +368,147 @@ def __repr__(self):
         return self.__str__()
 
 
+class UserDataSharingConsentAudit(TimeStampedModel):
+    """
+    Object that exists to store the canonical state of whether a particular
+    user has given consent for their course data to be shared with a particular
+    single-sign-on identity provider
+    """
+    class Meta(object):
+        app_label = 'enterprise'
+        verbose_name = "Data Sharing Consent Audit State"
+        verbose_name_plural = "Data Sharing Consent Audit States"
+        unique_together = ('user', 'enterprise_customer',)
+
+    NOT_SET = 'not_set'
+    ENABLED = 'enabled'
+    DISABLED = 'disabled'
+    STATE_CHOICES = (
+        (NOT_SET, 'Not set'),
+        (ENABLED, 'Enabled'),
+        (DISABLED, 'Disabled'),
+    )
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+    )
+
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        on_delete=models.CASCADE,
+    )
```

Comment:
I think you should replace `user` and `enterprise_customer` with a ForeignKey to `EnterpriseCustomerUser` instead.

If there's some technical reason you want to keep them separate, `user` should not be a `ForeignKey` for consistency with the `user` field of `EnterpriseCustomerUser`. But I think an FK to `EnterpriseCustomerUser` makes way more sense here.